### PR TITLE
Add f32_to_i2 and i2_to_f32 e2e tests.

### PR DESCRIPTION
**Pull request by hanhanW**

### CODE REVIEW

File: tests/e2e/linalg/fp_to_subbyte.mlir [hunk 1]
```diff
@@ -0,0 +1,49 @@
+func.func @f32_to_i4_1d() {
```
Comment (MaheshRavishankar) [inline tests/e2e/linalg/fp_to_subbyte.mlir:1]: Total nit: If you do a `git rename` before making changes, git will track it as a rename and is easier to see diff and maintain history.

**Review by MaheshRavishankar — APPROVED** [review]

File: tests/e2e/linalg/fp_to_subbyte.mlir (hunk 1 above)
Comment (hanhanW) [reply tests/e2e/linalg/fp_to_subbyte.mlir:1]: I did run `git mv`, but I don't remember if I did change first or I ran `git mv` first. I didn't know that the order matters, noted.

```
(base) [17:06:26] hanchung at hanhan-cpu in ~/iree/tests/e2e/linalg add-i2-tests
❯ git mv f32_to_i4.mlir fp_to_subbyte.mlir
(base) [17:07:18] hanchung at hanhan-cpu in ~/iree/tests/e2e/linalg add-i2-tests
❯ git mv i4_to_f32.mlir subbyte_to_fp.mlir
```

Comment (MaheshRavishankar) [reply tests/e2e/linalg/fp_to_subbyte.mlir:1]: Huh, id expect both to work, but I do `git mv` first.

File: tests/e2e/linalg/BUILD.bazel [hunk 2]
```diff
@@ -35,6 +35,10 @@ iree_check_single_backend_test_suite(
     name = "check_llvm-cpu_local-task",
     srcs = LLVM_SRCS,
     driver = "local-task",
+    tags = [
+        # subbyte support for wasm is not on priorities.
+        "nowasm",
+    ],
```
Comment (hanhanW) [inline tests/e2e/linalg/BUILD.bazel:41]: @ScottTodd I want to disable the test suite for wasm, but I failed. How do I disable it properly?

https://github.com/openxla/iree/actions/runs/8745270346/job/24001316668?pr=17074

Comment (ScottTodd) [reply tests/e2e/linalg/BUILD.bazel:41]: Hmmmm... I think this is only disabling the runtime portion (which doesn't exist on wasm anyways): https://github.com/openxla/iree/blob/f755b422892bfb8ebb2ee8258f9aac140cddbeb4/build_tools/cmake/iree_check_test.cmake#L135-L143

CMake configuration whack-a-mole... we could change that logic again, or we could set `-DIREE_BUILD_ALL_CHECK_TEST_MODULES=OFF` in the cross compile scripts. Let me try that and confirm that the expected sets of tests are still compiling/running. Looking at the logs, the emscripten build is compiling CUDA tests, when it really doesn't need to :P

Comment (ScottTodd) [reply tests/e2e/linalg/BUILD.bazel:41]: Ah, it's end of day actually, and tomorrow might be busy with the organization move. How urgent is this?

I'd basically want to set that option in https://github.com/openxla/iree/blob/main/build_tools/cmake/build_runtime_emscripten.sh (and https://github.com/openxla/iree/blob/main/build_tools/cmake/build_riscv.sh) like how it is already set in https://github.com/openxla/iree/blob/main/build_tools/cmake/build_android.sh, then look at the CI logs to see that all check tests for those targets are still being built.

Then for this PR, setting nowasm should I think exclude this particular list of tests again.

Comment (hanhanW) [reply tests/e2e/linalg/BUILD.bazel:41]: It is not urgent. I can wait. Thanks for taking a look at it!

Comment (ScottTodd) [reply tests/e2e/linalg/BUILD.bazel:41]: I think https://github.com/iree-org/iree/pull/17114 will help (without sacrificing too much coverage...)

**Pull request merged**